File: docs/examples/rs/examples/data_querying_filter.rs
use futures::StreamExt;
use reduct_rs::{ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a client instance, then get or create a bucket
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();
    let bucket = client
        .create_bucket("example-bucket")
        .exist_ok(true)
        .send()
        .await?;

    // Query 10 photos from "imdb" entry which taken in 2006 but don't contain "Rowan Atkinson"
    let query = bucket
        .query("imdb")
        .add_include("photo_taken", "2006")
        .add_exclude("name", "b'Rowan Atkinson'")
        .limit(10)
        .send()
        .await?;

    tokio::pin!(query);
    while let Some(record) = query.next().await {
        let record = record?;
        println!("Name: {:?}", record.labels().get("name"));
        println!("Photo Taken: {:?}", record.labels().get("photo_taken"));
        println!("Gender: {:?}", record.labels().get("gender"));

        _ = record.bytes().await?;
    }

    Ok(())
}

File: versioned_docs/version-1.20.x/examples/go/src/data_querying_filter.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get a bucket with the name "example-bucket"
	bucket, err := client.GetBucket(context.Background(), "example-bucket")
	if err != nil {
		panic(err)
	}

	// Query 10 photos from "imdb" entry which taken after 2006 with the face score less than 4
	queryOptions := reduct.NewQueryOptionsBuilder().
		WithWhen(map[string]any{
			"&photo_taken": map[string]any{"$gt": 2006},
			"&face_score":  map[string]any{"$lt": 4},
			"$limit":       10,
		}).
		Build()

	query, err := bucket.Query(context.Background(), "imdb", &queryOptions)
	if err != nil {
		panic(err)
	}

	for record := range query.Records() {
		println("Name:", record.Labels()["name"].(string))
		println("Photo taken:", record.Labels()["photo_taken"].(string))
		println("Face score:", record.Labels()["face_score"].(string))
	}
}
